membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -36,12 +36,37 @@ class AccountController extends LazyLogging {
   lazy val mmaAction = NoCacheAction andThen mmaCorsFilter andThen BackendFromCookieAction
   lazy val mmaCardAction = NoCacheAction andThen corsCardFilter andThen BackendFromCookieAction
 
+
+  def cancelContribution [P <: SubscriptionPlan.AnyPlan : SubPlanReads] = mmaCardAction.async { implicit request =>
+    val tp = request.touchpoint
+    val cancelForm = Form { single("reason" -> nonEmptyText) }
+
+    val maybeUserId = authenticationService.userId
+    logger.info(s"Attempting to cancel account for $maybeUserId")
+
+    (for {
+      user <- EitherT(Future.successful( maybeUserId \/> "no identity cookie for user"))
+      sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
+      zuoraSubscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
+      _ <- EitherT(tp.zuoraRestService.disableAutoPay(zuoraSubscription.accountId)).leftMap(message => s"Error message: $message")
+      _ <- EitherT(tp.zuoraRestService.cancelSubscription(zuoraSubscription.name)).leftMap(message => s"Error message: $message")
+      updateReasonResult <- EitherT(tp.zuoraRestService.updateCancellationReason(zuoraSubscription.name, cancelForm.bindFromRequest().value.get)).leftMap(message => s"Error message: $message")
+    } yield updateReasonResult).run.map {
+      case -\/(message) =>
+        logger.warn(s"Failed to update card for user $maybeUserId, due to $message")
```

Comment:
the name 'updateReasonResult' and the log message seem to be left over from the code you based this on right?